docs/source/point-evaluation.rst

```diff
@@ -104,6 +100,261 @@ in parallel. There is no special API, but there are some restrictions:
+
+
+Parallel behaviour
+~~~~~~~~~~~~~~~~~~
+
+In parallel the ``points`` given to :func:`~.VertexOnlyMesh` are assumed to be
+the same on each MPI process and are taken from rank 0. To let different ranks
+provide different points to the vertex-only mesh set the keyword argument
+``redundant = False``
+
+.. code-block:: python3
+
+   # Default behaviour
+   vom = VertexOnlyMesh(parent_mesh, points, redundant = True)
+
+   # Different points on each MPI rank to add to the vertex-only mesh
+   vom = VertexOnlyMesh(parent_mesh, points, redundant = False)
+
+In this case, ``points`` will redistribute to the mesh partition where they are
+located. This means that if rank A has ``points`` :math:`\{X\}` that are not
+found in the mesh cells owned by rank A but are found in the mesh cells owned
+by rank B then they will be moved to rank B.
+
+If the same coordinates are supplied more than once, they are always assumed to
+be a new vertex: this is true for both ``redundant = True`` and
+``redunant = False``. So if we have the same set of points on all MPI processes
+and switch from ``redundant = True`` to ``redundant = False`` we will get point
+duplication.
+
+
+Points outside the domain
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Be default points outside the domain  by more than the :ref:`specified
+tolerance <tolerance>` will generate a ``ValueError``. This can be switched
+to a warning or switched off entirely:
+
+.. literalinclude:: ../../tests/vertexonly/test_vertex_only_manual.py
+   :language: python3
+   :dedent:
+   :lines: 31-35,37-44
+
+
+Expressions with point evaluations
+----------------------------------
+
+Integrating over a vertex-only mesh is equivalent to summing over
+it. So if we have a vertex-only mesh :math:`\Omega_v` with :math:`N` vertices
+at points :math:`\{x_i\}_{i=0}^{N-1}` and we have interpolated a function
+:math:`f` onto it giving a new function :math:`f_v` then
+
+.. math::
+
+   \int_{\Omega_v} f_v \, dx = \sum_{i=0}^{N-1} f(x_i).
+
+These equivalent expressions for point evaluation
+
+.. math::
+
+   \sum_{i=0}^{N-1} f(x_i) = \sum_{i=0}^{N-1} \int_\Omega f(x) \delta(x - x_i) \, dx
+
+where :math:`N` is the number of points, :math:`x_i` is the :math:`i`\th point,
+:math:`\Omega` is a 'parent' mesh, :math:`f` is a function on that mesh,
+:math:`\delta` is a dirac delta distribition can therefore be written in
+Firedrake using :func:`~.VertexOnlyMesh` and :func:`~.interpolate` as
+
+.. literalinclude:: ../../tests/vertexonly/test_vertex_only_manual.py
+   :language: python3
+   :dedent:
+   :lines: 48-62
+
+Interacting with external point data
+------------------------------------
+
+Any set of points with associated data in our domain can be expressed as a
+P0DG function on a :func:`~.VertexOnlyMesh`.
+
+.. code-block:: python3
+
+   vom = VertexOnlyMesh(parent_mesh, point_locations_from_elsewhere)
+   P0DG = FunctionSpace(vom, "DG", 0)
+   y_pts = Function(P0DG).dat.data[:] = point_data_values_from_elsewhere
```

Comment:
I think we now know this isn't parallel-safe. Maybe we should drop this pending the next fix....

Reply:
TODO: add a warning that the example uses an internal firedrake API (`Function(P0DG).dat.data[:]`) and that it doesn’t work in parallel. I can state that a parallel safe way of interacting with external data will be available soon.